server/sonar-process-monitor/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -0,0 +1,287 @@
+      // just in case
+      if (process != null) {
+        process.destroyForcibly();
+      }
+      throw new IllegalStateException(format("Fail to launch process [%s]", esCommand.getProcessId().getKey()), e);
+    }
+  }
+
+  private void writeConfFiles(EsCommand esCommand) {
+    File confDir = esCommand.getConfDir();
+    if (!confDir.exists() && !confDir.mkdirs()) {
+      String error = format("Failed to create temporary configuration directory [%s]", confDir.getAbsolutePath());
+      LOG.error(error);
+      throw new IllegalStateException(error);
+    }
+
+    try {
+      IOUtils.copy(getClass().getResourceAsStream("elasticsearch.yml"), new FileOutputStream(new File(confDir, "elasticsearch.yml")));
+      IOUtils.copy(getClass().getResourceAsStream("jvm.options"), new FileOutputStream(new File(confDir, "jvm.options")));
+      esCommand.getLog4j2Properties().store(new FileOutputStream(new File(confDir, "log4j2.properties")), "log42 properties file for ES bundled in SonarQube");
+    } catch (IOException e) {
+      e.printStackTrace();
+      throw new IllegalStateException("Failed to write ES configuration files", e);
+    }
+  }
+
+  @Override
+  public ProcessMonitor launch(JavaCommand javaCommand) {
+    Process process = null;
+    try {
+      ProcessCommands commands = allProcessesCommands.createAfterClean(javaCommand.getProcessId().getIpcIndex());
+
+      ProcessBuilder processBuilder = create(javaCommand);
+      LOG.info("Launch process[{}]: {}", javaCommand.getProcessId().getKey(), String.join(" ", processBuilder.command()));
+      process = processBuilder.start();
+      return new ProcessCommandsProcessMonitor(process, commands);
+    } catch (Exception e) {
+      // just in case
+      if (process != null) {
+        process.destroyForcibly();
+      }
+      throw new IllegalStateException(format("Fail to launch process [%s]", javaCommand.getProcessId().getKey()), e);
+    }
+  }
+
+  private ProcessBuilder create(EsCommand esCommand) {
+    List<String> commands = new ArrayList<>();
+    commands.add(esCommand.getExecutable().getAbsolutePath());
+    commands.addAll(esCommand.getEsOptions());
+
+    writeJvmOptions(esCommand);
+
+    return create(esCommand, commands);
+  }
+
+  private void writeJvmOptions(EsCommand esCommand) {
+    Path jvmOptionsFile = esCommand.getJvmOptionsFile()
+      ;
+    LOG.warn("Daniels Debugging: writeJvmOptions ... "+jvmOptionsFile.toAbsolutePath());
+
+    String jvmOptions = esCommand.getJvmOptions()
+      .stream()
+      .map(s -> s.split(" (?=-)"))// FIXME this pattern does not allow escaping
+      .flatMap(Arrays::stream)
+      .collect(Collectors.joining("\n"));
+    String jvmOptionsContent = ELASTICSEARCH_JVM_OPTIONS_HEADER + jvmOptions;
+    try {
+      Files.write(jvmOptionsFile, jvmOptionsContent.getBytes(Charset.forName("UTF-8")));
+    } catch (IOException e) {
+      throw new RuntimeException("Cannot write Elasticsearch jvm options file", e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)